src/library/scala/collection/IndexedSeqView.scala

```diff
@@ -82,18 +83,18 @@ object IndexedSeqView {
     }
   }
   @SerialVersionUID(3L)
-  private final class IndexedSeqViewReverseIterator[A](self: IndexedSeqView[A]) extends AbstractIterator[A] with Serializable {
-    private[this] var pos = self.size - 1
-    private[this] var remainder = self.size
-    def hasNext: Boolean = remainder > 0
+  private[collection] class IndexedSeqViewReverseIterator[A](self: IndexedSeqView[A]) extends AbstractIterator[A] with Serializable {
+    private[this] var pos = self.length - 1
+    private[this] var remainder = self.length
+    @inline private[this] def _hasNext: Boolean = remainder > 0
+    def hasNext: Boolean = _hasNext
     def next(): A =
-      if (pos < 0) throw new NoSuchElementException
```

Comment:
this was a bug/mistake, btw. it should have been checking `remainder`, not `pos`